src/Components/CreateImageWizardV2/steps/TargetEnvironment/Azure/AzureSourcesSelect.tsx

```diff
@@ -31,51 +33,54 @@ const AzureSourcesSelect = ({ label, isRequired, className, ...props }) => {
     isError,
     refetch,
   } = useGetSourceListQuery({ provider: 'azure' });
-  const sources = extractProvisioningList(rawSources);
 
   const {
     data: sourceDetails,
     isFetching: isFetchingDetails,
     isSuccess: isSuccessDetails,
     isError: isErrorDetails,
   } = useGetSourceUploadInfoQuery(
-    { id: selectedSourceId },
+    { id: parseInt(azureSource as string) },
     {
-      skip: !selectedSourceId,
+      skip: !azureSource,
     }
   );
 
   useEffect(() => {
     if (isFetchingDetails || !isSuccessDetails) return;
-    change('azure-tenant-id', sourceDetails?.azure?.tenant_id);
-    change('azure-subscription-id', sourceDetails?.azure?.subscription_id);
+    dispatch(changeAzureTenantId(sourceDetails?.azure?.tenant_id || ''));
+    dispatch(
+      changeAzureSubscriptionId(sourceDetails?.azure?.subscription_id || '')
+    );
   }, [
     isFetchingDetails,
     isSuccessDetails,
-    sourceDetails?.azure?.subscription_id,
     sourceDetails?.azure?.tenant_id,
-    change,
+    sourceDetails?.azure?.subscription_id,
+    dispatch,
   ]);
 
-  const onFormChange = ({ values }) => {
-    if (
-      values['azure-type'] !== 'azure-type-source' ||
-      values[input.name] === undefined
-    ) {
-      change(input.name, undefined);
-      change('azure-tenant-id', undefined);
-      change('azure-subscription-id', undefined);
-    }
-  };
-
-  const handleSelect = (_, sourceName) => {
-    const sourceId = sources.find((source) => source.name === sourceName).id;
-    change(input.name, sourceId);
+  const handleSelect = (
+    _event: React.MouseEvent<Element, MouseEvent>,
+    sourceName: string
+  ) => {
+    const sourceId = rawSources?.data?.find(
+      (source) => source?.name === sourceName
+    )?.id;
+    dispatch(changeAzureSource(sourceId));
```

Comment:
I suppose we would need to add `|| ''` here if we change the type of `sourceId` to be `string` only in wizardSlice.